src/Altinn.App.Core/Internal/App/AppMetadata.cs

```diff
@@ -0,0 +1,112 @@
+using System.Text;
+using System.Text.Json;
+using Altinn.App.Core.Configuration;
+using Altinn.App.Core.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.App.Core.Internal.App
+{
+    /// <summary>
+    /// Default implementation of IAppMetadata
+    /// </summary>
+    public class AppMetadata : IAppMetadata
+    {
+        private readonly ILogger<AppMetadata> _logger;
+        private readonly AppSettings _settings;
+        private readonly IFrontendFeatures _frontendFeatures;
+        private ApplicationMetadata? _application;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AppMetadata"/> class.
+        /// </summary>
+        /// <param name="settings">The app repository settings.</param>
+        /// <param name="frontendFeatures">Application features service</param>
+        /// <param name="logger">A logger from the built in logger factory.</param>
+        public AppMetadata(
+            IOptions<AppSettings> settings,
+            IFrontendFeatures frontendFeatures,
+            ILogger<AppMetadata> logger)
+        {
+            _settings = settings.Value;
+            _frontendFeatures = frontendFeatures;
+            _logger = logger;
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="System.Text.Json.JsonException">Thrown if deserialization fails</exception>
+        /// <exception cref="System.IO.FileNotFoundException">Thrown if applicationmetadata.json file not found</exception>
+        public async Task<ApplicationMetadata> GetApplicationMetadata()
+        {
+            // Cache application metadata
+            if (_application != null)
+            {
+                return _application;
+            }
+
+            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.ApplicationMetadataFileName;
```

Comment:
Not you code, but a Path.Join would be good here.

Reply:
Nice catch, replacing all similar lines in AppMetadata with Path.Join